src/include/mpir_memtype.h

```diff
@@ -0,0 +1,18 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPIR_MEMTYPE_H_INCLUDED
+#define MPIR_MEMTYPE_H_INCLUDED
+
+typedef enum MPIR_Memtype {
+    MPIR_MEMTYPE__NONE = -1,
+    MPIR_MEMTYPE__DDR = 0,
+    MPIR_MEMTYPE__MCDRAM,
+    MPIR_MEMTYPE__NUM,
+    MPIR_MEMTYPE__DEFAULT = MPIR_MEMTYPE__DDR
```

Comment:
Is that allowed in C99?

Reply:
It should be allowed. The C99 standard does not say you can't do it (`section 6.7.2.2`):
`The expression that defines the value of an enumeration constant shall be an integer
constant expression that has a value representable as an int. [...] (The use of enumerators with
= may produce enumeration constants with values that duplicate other values in the same
enumeration.)`. My interpretation might be wrong though.